framework/sandstone.cpp

```diff
@@ -1990,6 +1990,37 @@ static void analyze_test_failures(int tc, const struct test *test, int fail_coun
         // can't use this information
         if (all_cpus_failed_equally)
             logging_printf(LOG_LEVEL_VERBOSE(1), "# All failing CPUs failed equally.\n");
+    } else if (test->flags & test_failure_package_only) {
+        // Failure cannot be attributed to a single thread or core.  Let's see if it
+        // can be pinned down to a single package.
+        logging_printf(LOG_LEVEL_VERBOSE(1), "# Topology analysis:\n");
+        std::vector<int> pkg_failures(topology.packages.size(), -1);
```

Comment:
I ponder if it makes sense to add fast-fail for 1 package case.

Reply:
Yeah, that's not a bad point.  The code here will work for single package systems, but I suppose that there's no point in iterating through the threads if we don't need to, particularly as this will be quite a common occurrence.  I'll add the fast path.